cert mixup

Something went wrong when installing the new certificates. This is the log output on the live owner app server

```
26.8.16 16:39:34.043 [Camel (wonOwnerCamel) thread #13 - TemporaryQueueReplyManager[OwnerProtocol.in]] WARN  o.a.c.c.j.r.TemporaryQueueReplyManager - Exception inside the DMLC for Temporary ReplyTo Queue for destination OwnerProtocol.in, refreshing ReplyTo destination
javax.jms.JMSException: Could not connect to broker URL: ssl://www.matchat.org:61617. Reason: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path validation failed: java.security.cert.CertPathValidatorException: Path does not chain with any of the trust anchors
        at org.apache.activemq.util.JMSExceptionSupport.create(JMSExceptionSupport.java:35) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:293) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:238) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createConnection(ActiveMQConnectionFactory.java:184) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.springframework.jms.connection.SingleConnectionFactory.doCreateConnection(SingleConnectionFactory.java:351) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.connection.SingleConnectionFactory.initConnection(SingleConnectionFactory.java:295) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.connection.SingleConnectionFactory.createConnection(SingleConnectionFactory.java:227) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.support.JmsAccessor.createConnection(JmsAccessor.java:180) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.AbstractJmsListeningContainer.createSharedConnection(AbstractJmsListeningContainer.java:413) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.AbstractJmsListeningContainer.refreshSharedConnection(AbstractJmsListeningContainer.java:398) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer.refreshConnectionUntilSuccessful(DefaultMessageListenerContainer.java:907) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer.recoverAfterListenerSetupFailure(DefaultMessageListenerContainer.java:882) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer$AsyncMessageListenerInvoker.run(DefaultMessageListenerContainer.java:1053) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_66-internal]
Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path validation failed: java.security.cert.CertPathValidatorException: Path does not chain with any of the trust anchors
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296) ~[na:1.8.0_66-internal]
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1509) ~[na:1.8.0_66-internal]
        at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:979) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.process_record(Handshaker.java:914) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.writeRecord(SSLSocketImpl.java:747) ~[na:1.8.0_66-internal]
        at sun.security.ssl.AppOutputStream.write(AppOutputStream.java:123) ~[na:1.8.0_66-internal]
        at org.apache.activemq.transport.tcp.TcpBufferedOutputStream.flush(TcpBufferedOutputStream.java:115) ~[activemq-client-5.8.0.jar:5.8.0]
        at java.io.DataOutputStream.flush(DataOutputStream.java:123) ~[na:1.8.0_66-internal]
        at org.apache.activemq.transport.tcp.TcpTransport.oneway(TcpTransport.java:176) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.AbstractInactivityMonitor.doOnewaySend(AbstractInactivityMonitor.java:322) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.AbstractInactivityMonitor.oneway(AbstractInactivityMonitor.java:304) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.WireFormatNegotiator.sendWireFormat(WireFormatNegotiator.java:168) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.WireFormatNegotiator.sendWireFormat(WireFormatNegotiator.java:84) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.WireFormatNegotiator.start(WireFormatNegotiator.java:74) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:273) ~[activemq-client-5.8.0.jar:5.8.0]
        ... 12 common frames omitted
Caused by: sun.security.validator.ValidatorException: PKIX path validation failed: java.security.cert.CertPathValidatorException: Path does not chain with any of the trust anchors
        at sun.security.validator.PKIXValidator.doValidate(PKIXValidator.java:352) ~[na:1.8.0_66-internal]
        at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:260) ~[na:1.8.0_66-internal]
        at sun.security.validator.Validator.validate(Validator.java:260) ~[na:1.8.0_66-internal]
        at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324) ~[na:1.8.0_66-internal]
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229) ~[na:1.8.0_66-internal]
        at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:105) ~[na:1.8.0_66-internal]
        at won.cryptography.ssl.TrustManagerWrapperWithTrustService.checkServerTrusted(TrustManagerWrapperWithTrustService.java:55) ~[won-core-0.2-SNAPSHOT.jar:na]
        at sun.security.ssl.AbstractTrustManagerWrapper.checkServerTrusted(SSLContextImpl.java:922) ~[na:1.8.0_66-internal]
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1491) ~[na:1.8.0_66-internal]
        ... 30 common frames omitted
Caused by: java.security.cert.CertPathValidatorException: Path does not chain with any of the trust anchors
        at sun.security.provider.certpath.PKIXCertPathValidator.validate(PKIXCertPathValidator.java:153) ~[na:1.8.0_66-internal]
        at sun.security.provider.certpath.PKIXCertPathValidator.engineValidate(PKIXCertPathValidator.java:79) ~[na:1.8.0_66-internal]
        at java.security.cert.CertPathValidator.validate(CertPathValidator.java:292) ~[na:1.8.0_66-internal]
        at sun.security.validator.PKIXValidator.doValidate(PKIXValidator.java:347) ~[na:1.8.0_66-internal]
        ... 38 common frames omitted
26.8.16 16:39:34.044 [Camel (wonOwnerCamel) thread #13 - TemporaryQueueReplyManager[OwnerProtocol.in]] ERROR o.a.c.c.j.DefaultJmsMessageListenerContainer - Could not refresh JMS Connection for destination 'temporary' - retrying using FixedBackOff{interval=5000, currentAttempts=253, maxAttempts=unlimited}. Cause: Could not connect to broker URL: ssl://www.matchat.org:61617. Reason: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path validation failed: java.security.cert.CertPathValidatorException: Path does not chain with any of the trust anchors
```

dis as described, now wonnode uses letsencrypt certificate, owner (matcher??) still uses self signed certificate. now we have a new error in wonnode:

31.8.16 08:55:30.582 [ActiveMQ Transport: ssl:///192.168.120.125:50594] WARN  o.a.a.b.T.Transport - Transport Connection to: tcp://192.168.120.125:50594 failed: javax.net.ssl.SSLHandshakeException: Received fatal alert: certificate_unknown
31.8.16 08:55:30.589 [ActiveMQ BrokerService[wonBroker] Task-9] ERROR o.a.a.broker.TransportConnector - Could not accept connection from tcp://192.168.120.125:50594: javax.net.ssl.SSLHandshakeException: Received fatal alert: certificate_unknown

and in owner: 
javax.jms.JMSException: Could not connect to broker URL: ssl://www.matchat.org:61616. Reason: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at org.apache.activemq.util.JMSExceptionSupport.create(JMSExceptionSupport.java:35) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:293) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:238) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createConnection(ActiveMQConnectionFactory.java:184) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.springframework.jms.connection.SingleConnectionFactory.doCreateConnection(SingleConnectionFactory.java:351) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.connection.SingleConnectionFactory.initConnection(SingleConnectionFactory.java:295) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.connection.SingleConnectionFactory.createConnection(SingleConnectionFactory.java:227) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.support.JmsAccessor.createConnection(JmsAccessor.java:180) ~[spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.AbstractJmsListeningContainer.createSharedConnection(AbstractJmsListeningContainer.java:413) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.AbstractJmsListeningContainer.establishSharedConnection(AbstractJmsListeningContainer.java:381) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer.establishSharedConnection(DefaultMessageListenerContainer.java:786) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.AbstractJmsListeningContainer.doStart(AbstractJmsListeningContainer.java:285) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.AbstractJmsListeningContainer.start(AbstractJmsListeningContainer.java:270) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.springframework.jms.listener.DefaultMessageListenerContainer.start(DefaultMessageListenerContainer.java:587) [spring-jms-4.1.1.RELEASE.jar:4.1.1.RELEASE]
        at org.apache.camel.component.jms.reply.ReplyManagerSupport.doStart(ReplyManagerSupport.java:229) [camel-jms-2.12.1.jar:2.12.1]
        at org.apache.camel.support.ServiceSupport.start(ServiceSupport.java:61) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:65) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.util.ServiceHelper.startService(ServiceHelper.java:54) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.component.jms.JmsProducer.createReplyManager(JmsProducer.java:516) [camel-jms-2.12.1.jar:2.12.1]
        at org.apache.camel.component.jms.JmsProducer.initReplyManager(JmsProducer.java:103) [camel-jms-2.12.1.jar:2.12.1]
        at org.apache.camel.component.jms.JmsProducer.processInOut(JmsProducer.java:177) [camel-jms-2.12.1.jar:2.12.1]
        at org.apache.camel.component.jms.JmsProducer.process(JmsProducer.java:142) [camel-jms-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.RedeliveryErrorHandler.process(RedeliveryErrorHandler.java:398) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:191) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.MulticastProcessor.doProcessSequential(MulticastProcessor.java:573) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.MulticastProcessor.doProcessSequential(MulticastProcessor.java:506) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.MulticastProcessor.process(MulticastProcessor.java:215) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.RecipientList.sendToRecipientList(RecipientList.java:153) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.RecipientList.process(RecipientList.java:112) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.Pipeline.process(Pipeline.java:118) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.Pipeline.process(Pipeline.java:80) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.management.InstrumentationProcessor.process(InstrumentationProcessor.java:72) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.RedeliveryErrorHandler.process(RedeliveryErrorHandler.java:398) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:191) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.ChoiceProcessor.process(ChoiceProcessor.java:103) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.management.InstrumentationProcessor.process(InstrumentationProcessor.java:72) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.RedeliveryErrorHandler.process(RedeliveryErrorHandler.java:398) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:191) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:191) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.component.seda.SedaConsumer.sendToConsumers(SedaConsumer.java:291) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.component.seda.SedaConsumer.doRun(SedaConsumer.java:200) [camel-core-2.12.1.jar:2.12.1]
        at org.apache.camel.component.seda.SedaConsumer.run(SedaConsumer.java:147) [camel-core-2.12.1.jar:2.12.1]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_66-internal]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_66-internal]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_66-internal]
Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296) ~[na:1.8.0_66-internal]
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1509) ~[na:1.8.0_66-internal]
        at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:216) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:979) ~[na:1.8.0_66-internal]
        at sun.security.ssl.Handshaker.process_record(Handshaker.java:914) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375) ~[na:1.8.0_66-internal]
        at sun.security.ssl.SSLSocketImpl.writeRecord(SSLSocketImpl.java:747) ~[na:1.8.0_66-internal]
        at sun.security.ssl.AppOutputStream.write(AppOutputStream.java:123) ~[na:1.8.0_66-internal]
        at org.apache.activemq.transport.tcp.TcpBufferedOutputStream.flush(TcpBufferedOutputStream.java:115) ~[activemq-client-5.8.0.jar:5.8.0]
        at java.io.DataOutputStream.flush(DataOutputStream.java:123) ~[na:1.8.0_66-internal]
        at org.apache.activemq.transport.tcp.TcpTransport.oneway(TcpTransport.java:176) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.AbstractInactivityMonitor.doOnewaySend(AbstractInactivityMonitor.java:322) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.AbstractInactivityMonitor.oneway(AbstractInactivityMonitor.java:304) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.WireFormatNegotiator.sendWireFormat(WireFormatNegotiator.java:168) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.WireFormatNegotiator.sendWireFormat(WireFormatNegotiator.java:84) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.WireFormatNegotiator.start(WireFormatNegotiator.java:74) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.TransportFilter.start(TransportFilter.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.ActiveMQConnectionFactory.createActiveMQConnection(ActiveMQConnectionFactory.java:273) ~[activemq-client-5.8.0.jar:5.8.0]
        ... 43 common frames omitted
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:387) ~[na:1.8.0_66-internal]
        at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:292) ~[na:1.8.0_66-internal]
        at sun.security.validator.Validator.validate(Validator.java:260) ~[na:1.8.0_66-internal]
        at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324) ~[na:1.8.0_66-internal]
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:229) ~[na:1.8.0_66-internal]
        at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:105) ~[na:1.8.0_66-internal]
        at won.cryptography.ssl.TrustManagerWrapperWithTrustService.checkServerTrusted(TrustManagerWrapperWithTrustService.java:55) ~[won-core-0.2-SNAPSHOT.jar:na]
        at sun.security.ssl.AbstractTrustManagerWrapper.checkServerTrusted(SSLContextImpl.java:922) ~[na:1.8.0_66-internal]
        at sun.security.ssl.ClientHandshaker.serverCertificate(ClientHandshaker.java:1491) ~[na:1.8.0_66-internal]
        ... 61 common frames omitted
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:146) ~[na:1.8.0_66-internal]
        at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:131) ~[na:1.8.0_66-internal]
        at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280) ~[na:1.8.0_66-internal]
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:382) ~[na:1.8.0_66-internal]
        ... 69 common frames omitted

and in matcher:
31.8.16 08:55:23.735 [Camel (ClusterSystem) thread #0 - JmsConsumer[MatcherProtocol.Out.Need]] ERROR o.a.c.c.j.DefaultJmsMessageListenerContainer - Could not refresh JMS Connection for destination 'MatcherProtocol.Out.Need' - retrying using FixedBackOff{interval=5000, currentAttempts=43, maxAttempts=unlimited}. Cause: Could not connect to broker URL: ssl://www.matchat.org:61616. Reason: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

--- a/webofneeds/won-docker/bigdata/Dockerfile
+++ b/webofneeds/won-docker/bigdata/Dockerfile
@@ -2,7 +2,7 @@
 # The bigdata-1.5.2-bundled.jar file has to be copied manually to this folder
 
 # use java as a base image
-FROM java:8
+FROM openjdk:8
 
 # download the big-data bundled jar
 ADD ./bigdata-1.5.2-bundled.jar /usr/src/bigdata/bigdata-bundled.jar

--- a/webofneeds/won-docker/bots/Dockerfile
+++ b/webofneeds/won-docker/bots/Dockerfile
@@ -2,7 +2,7 @@
 # directory have to be copied into this folder (done by maven build)
 
 # use java as a base image
-FROM java:8
+FROM openjdk:8
 
 # add webofneeds default config env variables
 ENV WON_CONFIG_DIR=/usr/src/bots/conf

--- a/webofneeds/won-docker/deploy_live_build.sh
+++ b/webofneeds/won-docker/deploy_live_build.sh
@@ -6,7 +6,7 @@ base_folder=/home/install
 
 #won_certificate_passwd=changeit
 #WORKSPACE=/var/lib/jenkins/jobs/webofneeds-integration-test/workspace
-#GENERATE_NEW_CERTIFICATES=false
+#GENERATE_NEW_CERTIFICATES=true
 #MAIL_USER=
 #MAIL_PASS=
 #MAIL_HOST=
@@ -72,7 +72,7 @@ docker -H satcluster01:2375 run --name=gencert -e CN="www.matchat.org" \
 # if it is not already started
 ssh root@satcluster02 mkdir -p $base_folder/won-server-certs
 mkdir -p ~/won-server-certs
-rm ~/won-server-certs/*
+rm -f ~/won-server-certs/*
 rsync root@satcluster01:$base_folder/won-server-certs/* ~/won-server-certs/
 rsync ~/won-server-certs/* root@satcluster02:$base_folder/won-server-certs/
 
@@ -157,6 +157,8 @@ docker -H satcluster02:2375 rm wonnode || echo 'No docker container found to rem
 docker -H satcluster02:2375 run --name=wonnode -d -e "uri.host=$public_node_uri" -e "http.port=8443" \
 -e "uri.prefix=https://${public_node_uri}/won" \
 -e "activemq.broker.port=61616" -p 8443:8443 -p 61616:61616 \
+-v $base_folder/letsencrypt/certs/live/www.matchat.org/fullchain.pem:/usr/local/tomcat/conf/ssl/t-cert.pem \
+-v $base_folder/letsencrypt/certs/live/www.matchat.org/privkey.pem:/usr/local/tomcat/conf/ssl/t-key.pem \
 -v $base_folder/letsencrypt/certs:/usr/local/tomcat/conf/ssl/ \
 -v $base_folder/won-client-certs/wonnode:/usr/local/tomcat/won/client-certs/ \
 -e "CERTIFICATE_PASSWORD=${won_certificate_passwd}" \
@@ -164,7 +166,6 @@ docker -H satcluster02:2375 run --name=wonnode -d -e "uri.host=$public_node_uri"
 -e "db.sql.jdbcDriverClass=org.postgresql.Driver" \
 -e "db.sql.jdbcUrl=jdbc:postgresql://satcluster02.researchstudio.at:5433/won_node" \
 -e "db.sql.user=won" -e "db.sql.password=won" \
--e "JMEM_OPTS=-Xmx400m -XX:MaxMetaspaceSize=200m -XX:+HeapDumpOnOutOfMemoryError" \
 webofneeds/wonnode:${deploy_image_tag_name}
 
 # bigdata
@@ -187,14 +188,12 @@ docker ${docker_options} run --name=matcher_service -d -e "node.host=satcluster0
 -e "wonNodeController.wonNode.crawl=https://${public_node_uri}/won/resource" \
 -e "cluster.local.port=2561" -e "cluster.seed.port=2561" -p 2561:2561 \
 -v $base_folder/won-client-certs/matcher_service:/usr/src/matcher-service/client-certs/ \
--e "JMEM_OPTS=-Xmx250m -XX:MaxMetaspaceSize=200m -XX:+HeapDumpOnOutOfMemoryError" \
 webofneeds/matcher_service:${deploy_image_tag_name}
 
 # solr server
 echo run solr server container
 docker ${docker_options} pull webofneeds/solr
-if ! docker ${docker_options} run --name=solr -d -p 7071:8080 -p 8984:8983 --env CATALINA_OPTS="-Xmx200m \
-     -XX:MaxPermSize=150m -XX:+HeapDumpOnOutOfMemoryError" webofneeds/solr; then
+if ! docker ${docker_options} run --name=solr -d -p 7071:8080 -p 8984:8983 webofneeds/solr; then
   echo solr server container already available, restart old container
   docker ${docker_options} restart solr
 fi
@@ -231,9 +230,7 @@ docker ${docker_options} run --name=matcher_solr -d -e "node.host=satcluster01.r
 -e "cluster.seed.host=satcluster01.researchstudio.at" -e "cluster.seed.port=2561" -e "cluster.local.port=2562" \
 -e "matcher.solr.uri.solr.server=http://satcluster01.researchstudio.at:8984/solr/" \
 -e "matcher.solr.uri.solr.server.public=http://satcluster01.researchstudio.at:8984/solr/" \
--p 2562:2562 -e "JMEM_OPTS=-Xmx250m -XX:MaxMetaspaceSize=200m -XX:+HeapDumpOnOutOfMemoryError" \
-webofneeds/matcher_solr:${deploy_image_tag_name}
-
+-p 2562:2562 webofneeds/matcher_solr:${deploy_image_tag_name}
 
 
 # =====================================================================

--- a/webofneeds/won-docker/deploy_live_build.sh
+++ b/webofneeds/won-docker/deploy_live_build.sh
@@ -4,6 +4,14 @@ set -e
 # base folder is used to mount some files (e.g. certificates) from the server into the containers
 base_folder=/home/install
 
+#won_certificate_passwd=changeit
+#WORKSPACE=/var/lib/jenkins/jobs/webofneeds-integration-test/workspace
+#GENERATE_NEW_CERTIFICATES=false
+#MAIL_USER=
+#MAIL_PASS=
+#MAIL_HOST=
+
+
 echo build webofneeds images and create certificates
 
 # if the GENERATE_NEW_CERTIFICATES flag is set to true then setup things in a way that the certificates are recreated
@@ -16,6 +24,8 @@ if [ "$GENERATE_NEW_CERTIFICATES" = true ] ; then
   ssh root@satcluster01 rm -rf $base_folder/won-client-certs/*
   docker -H satcluster01:2375 stop postgres || echo 'No docker container found to stop with name: postgres'
   docker -H satcluster01:2375 rm postgres || echo 'No docker container found to remove with name: postgres'
+  docker -H satcluster02:2375 stop postgres || echo 'No docker container found to stop with name: postgres'
+  docker -H satcluster02:2375 rm postgres || echo 'No docker container found to remove with name: postgres'
   docker -H satcluster01:2375 stop bigdata || echo 'No docker container found to stop with name: bigdata'
   docker -H satcluster01:2375 rm bigdata || echo 'No docker container found to remove with name: bigdata'
   docker -H satcluster01:2375 stop solr || echo 'No docker container found to stop with name: solr'
@@ -23,7 +33,7 @@ if [ "$GENERATE_NEW_CERTIFICATES" = true ] ; then
 fi
 
 echo start docker build of images:
-docker -H satcluster01:2375 build -t webofneeds/wonnode:live $WORKSPACE/webofneeds/won-docker/wonnode/
+docker -H satcluster02:2375 build -t webofneeds/wonnode:live $WORKSPACE/webofneeds/won-docker/wonnode/
 docker -H satcluster01:2375 build -t webofneeds/owner:live $WORKSPACE/webofneeds/won-docker/owner/
 docker -H satcluster01:2375 build -t webofneeds/matcher_service:live $WORKSPACE/webofneeds/won-docker/matcher-service/
 docker -H satcluster01:2375 build -t webofneeds/matcher_solr:live $WORKSPACE/webofneeds/won-docker/matcher-solr/
@@ -43,7 +53,7 @@ rm won_certificate_passwd_file
 # entries of alternative server names: www.matchat.org, satcluster01.researchstudio.at, satcluster02.researchstudio.at
 scp $WORKSPACE/webofneeds/won-docker/gencert/openssl.conf root@satcluster01:$base_folder/openssl.conf
 
-# wonnode/owner server certificate generator
+# owner server certificate generator
 # PLEASE NOTE that value of PASS should be the same used in your server.xml for SSLPassword on wonnode and owner,
 # and the same as activemq.broker.keystore.password used in your wonnode activemq spring configurations for broker
 # this image creates a keypair and a certificate based on it and stores the respective files as in the specified
@@ -74,14 +84,14 @@ rsync $WORKSPACE/webofneeds/won-docker/nginx/nginx-http-only.conf root@satcluste
 # start the letsencrypt ssl certificate request cron job container, executes cron job requests/renewals every 12 hours
 docker -H satcluster02:2375 stop letsencrypt || echo 'No docker container found to stop with name: letsencrypt'
 docker -H satcluster02:2375 rm letsencrypt || echo 'No docker container found to remove with name: letsencrypt'
-docker -H satcluster02:2375 run --name=letsencrypt -d -v /home/install/letsencrypt/acme-challenge:/usr/share/nginx/html/ \
--v /home/install/letsencrypt/certs:/etc/letsencrypt/ webofneeds/letsencrypt:live
+docker -H satcluster02:2375 run --name=letsencrypt -d -v $base_folder/letsencrypt/acme-challenge:/usr/share/nginx/html/ \
+-v $base_folder/letsencrypt/certs:/etc/letsencrypt/ -e "key_store_password=${won_certificate_passwd}" webofneeds/letsencrypt:live
 
 # bootstrap code: if no nginx is running yet start it up in htt-only mode to generate a first letsencrypt certificate.
 # Then shut it down and start the nginx in normal (ssl) mode with the letsencrypt certificate
 if docker -H satcluster02:2375 run --name=nginx_http_only \
--v /home/install/nginx-http-only.conf:/etc/nginx/nginx.conf \
--v /home/install/letsencrypt/acme-challenge:/usr/share/nginx/html/ -d -p 80:80 nginx; then
+-v $base_folder/nginx-http-only.conf:/etc/nginx/nginx.conf \
+-v $base_folder/letsencrypt/acme-challenge:/usr/share/nginx/html/ -d -p 80:80 nginx; then
   echo bootstrap: no nginx proxy was running on this server, started nginx in http-only mode and try to retrieve an letsencrypt certificate
 
   # start the letsencrypt container with CMD to execute the certificate challenge once
@@ -106,13 +116,133 @@ docker_options="-H satcluster01:2375"
 public_node_uri=www.matchat.org
 behind_proxy=true
 deploy_image_tag_name=live
-. $WORKSPACE/webofneeds/won-docker/deploy_live_run.sh
+
+
+# =====================================================================
+# deploy application containers
+# =====================================================================
+
+echo "deployment options: "
+echo "docker_options: $docker_options"
+echo "deploy_image_tag_name: $deploy_image_tag_name"
+echo "public_node_uri: $public_node_uri"
+echo "behind_proxy: $behind_proxy"
+echo "base_folder: $base_folder"
+
+# start the won containers on dedicated servers of the cluster
+echo run docker containers:
+
+# postgres
+# NOTE: do not redeploy the postgres database for won node and owner to keep the data after deployments postgres db
+echo run postgres containers
+docker -H satcluster01:2375 pull webofneeds/postgres
+if ! docker -H satcluster01:2375 run --name=postgres -d -p 5433:5432 webofneeds/postgres; then
+  echo postgres container already available, restart old container
+  docker -H satcluster01:2375 restart postgres
+fi
+
+docker -H satcluster02:2375 pull webofneeds/postgres
+if ! docker -H satcluster02:2375 run --name=postgres -d -p 5433:5432 webofneeds/postgres; then
+  echo postgres container already available, restart old container
+  docker -H satcluster02:2375 restart postgres
+fi
+
+
+sleep 10
+
+# wonnode
+echo run wonnode container
+docker -H satcluster02:2375 stop wonnode || echo 'No docker container found to stop with name: wonnode'
+docker -H satcluster02:2375 rm wonnode || echo 'No docker container found to remove with name: wonnode'
+docker -H satcluster02:2375 run --name=wonnode -d -e "uri.host=$public_node_uri" -e "http.port=8443" \
+-e "uri.prefix=https://${public_node_uri}/won" \
+-e "activemq.broker.port=61616" -p 8443:8443 -p 61616:61616 \
+-v $base_folder/letsencrypt/certs:/usr/local/tomcat/conf/ssl/ \
+-v $base_folder/won-client-certs/wonnode:/usr/local/tomcat/won/client-certs/ \
+-e "CERTIFICATE_PASSWORD=${won_certificate_passwd}" \
+-e "client.authentication.behind.proxy=$behind_proxy" \
+-e "db.sql.jdbcDriverClass=org.postgresql.Driver" \
+-e "db.sql.jdbcUrl=jdbc:postgresql://satcluster02.researchstudio.at:5433/won_node" \
+-e "db.sql.user=won" -e "db.sql.password=won" \
+-e "JMEM_OPTS=-Xmx400m -XX:MaxMetaspaceSize=200m -XX:+HeapDumpOnOutOfMemoryError" \
+webofneeds/wonnode:${deploy_image_tag_name}
+
+# bigdata
+echo run bigdata container
+docker ${docker_options} pull webofneeds/bigdata
+if ! docker ${docker_options} run --name=bigdata -d -p 10000:9999 -m 400m webofneeds/bigdata; then
+  echo bigdata container already available, restart old container
+  docker ${docker_options} restart bigdata
+fi
+
+sleep 10
+
+# matcher service
+echo run matcher service container
+docker ${docker_options} stop matcher_service || echo 'No docker container found to stop with name: matcher_service'
+docker ${docker_options} rm matcher_service || echo 'No docker container found to remove with name: matcher_service'
+docker ${docker_options} run --name=matcher_service -d -e "node.host=satcluster01.researchstudio.at" \
+-e "cluster.seed.host=satcluster01.researchstudio.at" \
+-e "uri.sparql.endpoint=http://satcluster01.researchstudio.at:10000/bigdata/namespace/kb/sparql" \
+-e "wonNodeController.wonNode.crawl=https://${public_node_uri}/won/resource" \
+-e "cluster.local.port=2561" -e "cluster.seed.port=2561" -p 2561:2561 \
+-v $base_folder/won-client-certs/matcher_service:/usr/src/matcher-service/client-certs/ \
+-e "JMEM_OPTS=-Xmx250m -XX:MaxMetaspaceSize=200m -XX:+HeapDumpOnOutOfMemoryError" \
+webofneeds/matcher_service:${deploy_image_tag_name}
+
+# solr server
+echo run solr server container
+docker ${docker_options} pull webofneeds/solr
+if ! docker ${docker_options} run --name=solr -d -p 7071:8080 -p 8984:8983 --env CATALINA_OPTS="-Xmx200m \
+     -XX:MaxPermSize=150m -XX:+HeapDumpOnOutOfMemoryError" webofneeds/solr; then
+  echo solr server container already available, restart old container
+  docker ${docker_options} restart solr
+fi
+
+sleep 10
+
+# expect OWNER won-mail-sender host, user and password (i.e. configuration for no-replay won-owner-app-email-account) be
+# set as environment variables, e.g. MAIL_USER=changeuser MAIL_PASS=changepass MAIL_HOST=smtp.changehost.com
+echo ${MAIL_USER} at ${MAIL_HOST} is used as owner no-replay won-owner-app-email-account
+
+# owner
+echo run owner container
+docker ${docker_options} stop owner || echo 'No docker container found to stop with name: owner'
+docker ${docker_options} rm owner || echo 'No docker container found to remove with name: owner'
+docker ${docker_options} run --name=owner -d -e "node.default.host=$public_node_uri" \
+-e "node.default.http.port=443" -p 8082:8443 \
+-e "uri.host=$public_node_uri" -e "http.port=8082" \
+-e "uri.prefix=https://$public_node_uri" \
+-e "uri.prefix.node.default=https://${public_node_uri}/won" \
+-e "CERTIFICATE_PASSWORD=${won_certificate_passwd}" \
+-e "email.from.won.user=${MAIL_USER}" -e "email.from.won.password=${MAIL_PASS}" -e "email.from.won.smtp.host=${MAIL_HOST}" \
+-v $base_folder/won-server-certs:/usr/local/tomcat/conf/ssl/ \
+-v $base_folder/won-client-certs/owner:/usr/local/tomcat/won/client-certs/ \
+-e "db.sql.jdbcDriverClass=org.postgresql.Driver" \
+-e "db.sql.jdbcUrl=jdbc:postgresql://satcluster01.researchstudio.at:5433/won_owner" \
+-e "db.sql.user=won" -e "db.sql.password=won" \
+webofneeds/owner:${deploy_image_tag_name}
+
+# solr matcher
+echo run solr matcher container
+docker ${docker_options} stop matcher_solr || echo 'No docker container found to stop with name: matcher_solr'
+docker ${docker_options} rm matcher_solr || echo 'No docker container found to remove with name: matcher_solr'
+docker ${docker_options} run --name=matcher_solr -d -e "node.host=satcluster01.researchstudio.at" \
+-e "cluster.seed.host=satcluster01.researchstudio.at" -e "cluster.seed.port=2561" -e "cluster.local.port=2562" \
+-e "matcher.solr.uri.solr.server=http://satcluster01.researchstudio.at:8984/solr/" \
+-e "matcher.solr.uri.solr.server.public=http://satcluster01.researchstudio.at:8984/solr/" \
+-p 2562:2562 -e "JMEM_OPTS=-Xmx250m -XX:MaxMetaspaceSize=200m -XX:+HeapDumpOnOutOfMemoryError" \
+webofneeds/matcher_solr:${deploy_image_tag_name}
+
+
+
+# =====================================================================
 
 echo push automatically built webobofneeds images to docker hub
 docker -H satcluster01:2375 login -u heikofriedrich
 docker -H satcluster01:2375 push webofneeds/gencert:live
-docker -H satcluster01:2375 push webofneeds/wonnode:live
+docker -H satcluster02:2375 push webofneeds/wonnode:live
 docker -H satcluster01:2375 push webofneeds/owner:live
 docker -H satcluster01:2375 push webofneeds/matcher_service:live
 docker -H satcluster01:2375 push webofneeds/matcher_solr:live
-docker -H satcluster01:2375 push webofneeds/letsencrypt:live
+docker -H satcluster02:2375 push webofneeds/letsencrypt:live

--- a/webofneeds/won-docker/letsencrypt/Dockerfile
+++ b/webofneeds/won-docker/letsencrypt/Dockerfile
@@ -1,7 +1,18 @@
-FROM debian:jessie
+FROM openjdk:8
+
+# install certbot
 ADD ./jessie-backports.list /etc/apt/sources.list.d/jessie-backports.list
 RUN apt-get update && apt-get install -y certbot -t jessie-backports
+
+# install cron
 RUN apt-get install -y cron
+
+ENV key_store_password=changeit
+ENV out_folder=/etc/letsencrypt
+ENV key_pem_file="/etc/letsencrypt/live/www.matchat.org/privkey.pem"
+ENV cert_pem_file="/etc/letsencrypt/live/www.matchat.org/fullchain.pem"
+ENV pfx_store_file="/etc/letsencrypt/t-key-cert.pfx"
+ENV jks_store_file="/etc/letsencrypt/t-keystore.jks"
 
 VOLUME /etc/letsencrypt
 VOLUME /usr/share/nginx/html
@@ -9,8 +20,8 @@ VOLUME /usr/share/nginx/html
 ADD crontab-certificate-request-and-renew /etc/cron.d/crontab-certificate-request-and-renew
 ADD certificate-request-and-renew.sh /usr/local/bin/certificate-request-and-renew.sh
 
-RUN chmod 777 /etc/cron.d/crontab-certificate-request-and-renew
-RUN chmod 777 /usr/local/bin/certificate-request-and-renew.sh
+RUN chmod 744 /etc/cron.d/crontab-certificate-request-and-renew
+RUN chmod 744 /usr/local/bin/certificate-request-and-renew.sh
 
 RUN /usr/bin/crontab /etc/cron.d/crontab-certificate-request-and-renew
 RUN touch /var/log/cron.log

--- a/webofneeds/won-docker/letsencrypt/certificate-request-and-renew.sh
+++ b/webofneeds/won-docker/letsencrypt/certificate-request-and-renew.sh
@@ -1,1 +1,12 @@
-certbot certonly --webroot -w /usr/share/nginx/html -d www.matchat.org --email florian.kleedorfer@researchstudio.at -t --noninteractive --agree-tos
+echo renew letsencrypt certificate
+certbot certonly --webroot -w /usr/share/nginx/html -d www.matchat.org --email florian.kleedorfer@researchstudio.at \
+-t --noninteractive --agree-tos
+
+# create pfx key store
+echo create pfx key store: $pfx_store_file
+openssl pkcs12 -export -out $pfx_store_file -passout pass:$key_store_password -inkey $key_pem_file -in $cert_pem_file
+
+# create java key store
+echo create java key store: $jks_store_file
+keytool -importkeystore -srckeystore $pfx_store_file -srcstoretype pkcs12 -destkeystore $jks_store_file \
+-deststoretype JKS -srcstorepass $key_store_password -deststorepass $key_store_password -noprompt

--- a/webofneeds/won-docker/letsencrypt/crontab-certificate-request-and-renew
+++ b/webofneeds/won-docker/letsencrypt/crontab-certificate-request-and-renew
@@ -1,4 +1,2 @@
 # try to renew the certificate every 12 hours at a random time (--dry-run can be used for testing)
-0 */12 * * * perl -e 'sleep int(rand(3600))' && /usr/local/bin/certificate-request-and-renew.sh >> /var/log/cron.log 2>&1
-
-
+0 */12 * * * perl -e 'sleep int(rand(3600))' && echo \ncronjob running at `date` >> /var/log/cron.log && /usr/local/bin/certificate-request-and-renew.sh >> /var/log/cron.log 2>&1

--- a/webofneeds/won-docker/matcher-service/Dockerfile
+++ b/webofneeds/won-docker/matcher-service/Dockerfile
@@ -2,7 +2,7 @@
 # conf directory have to be copied into this folder (done by maven build)
 
 # use java as a base image
-FROM java:8
+FROM openjdk:8
 
 # add webofneeds default config env variables
 ENV WON_CONFIG_DIR=/usr/src/matcher-service/conf

--- a/webofneeds/won-docker/matcher-solr/Dockerfile
+++ b/webofneeds/won-docker/matcher-solr/Dockerfile
@@ -2,7 +2,7 @@
 # directory have to be copied into this folder (done by maven build)
 
 # use java as a base image
-FROM java:8
+FROM openjdk:8
 
 # add webofneeds default config env variables
 ENV WON_CONFIG_DIR=/usr/src/matcher-solr/conf

--- a/webofneeds/won-docker/nginx/nginx.conf
+++ b/webofneeds/won-docker/nginx/nginx.conf
@@ -13,7 +13,7 @@ stream {
     # pass tcp jms messages
     server {
         listen 61617;
-        proxy_pass satcluster01.researchstudio.at:61617;
+        proxy_pass satcluster01.researchstudio.at:61616;
     }
 }
 
@@ -68,7 +68,7 @@ http {
         location /won {
             # set the client certificate in this header when the certificate was already validated by nginx
             proxy_set_header X-Client-Certificate $ssl_client_cert;
-            proxy_pass https://satcluster01.researchstudio.at:443/won;
+            proxy_pass https://satcluster02.researchstudio.at:8443/won;
         }
 
         location /owner {

--- a/webofneeds/won-docker/owner/Dockerfile
+++ b/webofneeds/won-docker/owner/Dockerfile
@@ -6,20 +6,11 @@ RUN apt-get update && apt-get install -y \
     less \
 	dos2unix
 
-# install oracle jre 8 latest version
-RUN echo "deb http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main" | tee /etc/apt/sources.list.d/webupd8team-java.list
-RUN "deb-src http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main" | tee -a /etc/apt/sources.list.d/webupd8team-java.list
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys EEA14886
-RUN apt-get update -y
-RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections
-RUN apt-get install oracle-java8-installer -y
-RUN apt-get install oracle-java8-set-default
-
 # compile APR library to use for tomcat ssl connector
-RUN cd /usr/local/tomcat/bin/
 RUN tar -xf /usr/local/tomcat/bin/tomcat-native.tar.gz -C /usr/local/tomcat/bin/
-RUN  apt-get update && apt-get install libapr1-dev libssl-dev gcc make -y
-RUN cd /usr/local/tomcat/bin/tomcat-native-1.1.33-src/jni/native/ && ./configure --with-apr=/usr --with-java-home=/usr/lib/jvm/java-8-oracle && make && make install
+RUN apt-get update && apt-get install libapr1-dev libssl-dev openjdk-8-jdk gcc make -y
+RUN cd /usr/local/tomcat/bin/$(ls /usr/local/tomcat/bin/ | grep tomcat-native-)/jni/native/ && \
+./configure --with-apr=/usr --with-java-home=/usr/lib/jvm/java-8-openjdk-amd64 && make && make install
 
 # add webofneeds default config env variables
 ENV WON_CONFIG_DIR=/usr/local/tomcat/won/conf

--- a/webofneeds/won-docker/wonnode/Dockerfile
+++ b/webofneeds/won-docker/wonnode/Dockerfile
@@ -6,20 +6,11 @@ RUN apt-get update && apt-get install -y \
     less \
 	dos2unix
 
-# install oracle jre 8 latest version
-RUN echo "deb http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main" | tee /etc/apt/sources.list.d/webupd8team-java.list
-RUN "deb-src http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main" | tee -a /etc/apt/sources.list.d/webupd8team-java.list
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys EEA14886
-RUN apt-get update -y
-RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | /usr/bin/debconf-set-selections
-RUN apt-get install oracle-java8-installer -y
-RUN apt-get install oracle-java8-set-default
-
 # compile APR library to use for tomcat ssl connector
 RUN tar -xf /usr/local/tomcat/bin/tomcat-native.tar.gz -C /usr/local/tomcat/bin/
-RUN apt-get update && apt-get install libapr1-dev libssl-dev gcc make -y
+RUN apt-get update && apt-get install libapr1-dev libssl-dev openjdk-8-jdk gcc make -y
 RUN cd /usr/local/tomcat/bin/$(ls /usr/local/tomcat/bin/ | grep tomcat-native-)/jni/native/ && \
-./configure --with-apr=/usr --with-java-home=/usr/lib/jvm/java-8-oracle && make && make install
+./configure --with-apr=/usr --with-java-home=/usr/lib/jvm/java-8-openjdk-amd64 && make && make install
 
 # add webofneeds default config env variables
 ENV WON_CONFIG_DIR=/usr/local/tomcat/won/conf

